variables-and-memory/problem-set-variables-and-memory.md

```diff
@@ -1 +1,487 @@
+3       input: word (string) and letter (single character)
+4       output: boolean
+5       """
+6       if letter not in word:
+7           result = False
+8       else: 
+9           result = True
+10
+11      return result
+12
+13 result = True
+14 result = is_letter_in_word("hello", "x"):
+```
+
+##### !end-question
+
+##### !options
+
+* `True`
+* `False`
+* `"hello"`
+* `"x"`
+
+##### !end-options
+
+##### !answer
+
+* `False`
+
+##### !end-answer
+
+##### !explanation 
+
+When we invoke `is_letter_in_word` on line `14` we assign its return value to the global variable `result`. As such, the value of `result` is overwritten with `False` (the value returned by the function).
+
+##### !end-explanation
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, hidden, students click to view) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+<!-- !explanation - !end-explanation (markdown, students can see after answering correctly) -->
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: c8a41151-3eb4-4a4f-ac8d-cab3b1e869c3
+* title: Variables Are References
+<!-- * points: [1] (optional, the number of points for scoring as a checkpoint) -->
+<!-- * topics: [python, pandas] (optional the topics for analyzing points) -->
+
+##### !question
+
+What are the values of `a` and `b` after the following code snippet is run?
+
+```py
+a = "hello"
+b = a
+a = "world"
+```
+
+##### !end-question
+
+##### !options
+
+* `a = "hello"`, `b = "hello"`
+* `a = "hello"`, `b = "world"`
+* `a = "world"`, `b = "hello"`
+* `a = "world"`, `b = "world"`
+
+##### !end-options
+
+##### !answer
+
+* `a = "world"`, `b = "hello"`
+
+##### !end-answer
+
+##### !explanation
+
+Strings are an immutable datatype. When `a` is reassigned to the value `"world"`, `b` still points to the value `"hello"`.
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: bf869b3d-23f9-4643-b1a4-9c661c865ac1
+* title: Variables Are References
+<!-- * points: [1] (optional, the number of points for scoring as a checkpoint) -->
+<!-- * topics: [python, pandas] (optional the topics for analyzing points) -->
+
+##### !question
+
+What are the values of `before_tax` and `after_tax` after the following code snippet is run?
+
+```py
+before_tax = 10
+after_tax = before_tax  * 1.1
+before_tax = 20
+```
+
+##### !end-question
+
+##### !options
+
+* `a = 10`, `b = 11`
```

Comment:
Name mismatch: variable names in the options & answer are `a` & `b` but in the question & code block are `before_tax` & `after_tax`

Reply:
I just caught that as I was reviewing your review. Thanks!